### PR TITLE
Fix #47 by explicitly using Error.message to generate Notifications

### LINKED ISSUE #47: Uncaught Error: Notification must be created with string message: Error: ENOENT: no such file or ...
[Enter steps to reproduce below:]

1. Install 'editorconfig' package;
2. Type `cmd+shift+p` to open the command palette;
3. Issue the `EditorConfig: Generate Config` command.

**Atom Version**: 1.0.7
**System**: Mac OS X 10.9.5
**Thrown From**: [editorconfig](https://github.com/sindresorhus/atom-editorconfig) package, v1.1.1

### Stack Trace

Uncaught Error: Notification must be created with string message: Error: ENOENT: no such file or directory, open 'atom:/.atom/keymap/.editorconfig'

```
At /Users/z418047/Downloads/Atom.app/Contents/Resources/app.asar/src/notification.js:25

Error: Notification must be created with string message: Error: ENOENT: no such file or directory, open 'atom:/.atom/keymap/.editorconfig'
  at Notification.module.exports.Notification.validate (/Users/z418047/Downloads/Atom.app/Contents/Resources/app.asar/src/notification.js:25:15)
  at new Notification (/Users/z418047/Downloads/Atom.app/Contents/Resources/app.asar/src/notification.js:20:12)
  at NotificationManager.module.exports.NotificationManager.addError (/Users/z418047/Downloads/Atom.app/Contents/Resources/app.asar/src/notification-manager.js:41:35)
  at /Users/z418047/.atom/packages/editorconfig/commands/generate.js:37:25
  at fs.js:1133:21
  at FSReqWrap.oncomplete (fs.js:77:15)

```

### Commands

```
     -8:43.9.0 editor:consolidate-selections (atom-text-editor.editor.mini.is-focused)
     -8:43.9.0 core:cancel (atom-text-editor.editor.mini.is-focused)
     -8:43.6.0 editor:consolidate-selections (atom-text-editor.editor.mini.is-focused)
     -8:43.6.0 core:cancel (atom-text-editor.editor.mini.is-focused)
     -8:26 core:close (atom-text-editor.editor.mini.is-focused)
     -8:24.3.0 settings-view:open (atom-text-editor.editor.is-focused)
     -8:13.1.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -8:11.5.0 editor:move-to-first-character-of-line (atom-text-editor.editor.mini.is-focused)
     -8:10.9.0 core:move-left (atom-text-editor.editor.mini.is-focused)
     -8:09.4.0 editor:move-to-first-character-of-line (atom-text-editor.editor.mini.is-focused)
     -8:08.9.0 editor:move-to-beginning-of-word (atom-text-editor.editor.mini.is-focused)
     -8:08.4.0 core:select-left (atom-text-editor.editor.mini.is-focused)
     -2:22.9.0 core:confirm (atom-text-editor.editor.mini.is-focused)
  2x -0:29.1.0 command-palette:toggle (atom-text-editor.editor.mini.is-focused)
     -0:00.7.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:00.7.0 EditorConfig:generate-config (atom-text-editor.editor.mini)
```

### Config

```json
{
  "core": {
    "themes": [
      "atom-dark-ui",
      "base16-tomorrow-dark-theme"
    ]
  }
}
```

### Installed Packages

```coffee
# User
editorconfig, v1.1.1

# Dev
No dev packages
```

### PR DESCRIPTION
**Pull request by florianb**
* fixes #47 
* fixes malformed tests
* ⚠️ this bug is untestable, since i wasn't able to open a TextEditor in a state which would provoke this Error.

### CODE REVIEW

**Comment by sindresorhus** [comment]
Seems like unrelated commits got included here, like https://github.com/sindresorhus/atom-editorconfig/pull/84/commits/a8d2e467f7e4e66faf58dba112e48a38a14c0737.

File: commands/generate.js [hunk 1]
```diff
@@ -46,7 +46,7 @@ trim_trailing_whitespace = false
 		if (err) {
 			fs.writeFile(configFile, ret, err => {
 				if (err) {
-					atom.notifications.addError(err);
+					atom.notifications.addError(err.message);
```
Comment (sindresorhus) [inline commands/generate.js]: Can you add the `err.stack` as the `detail` option?

What do you think about opening an issue on Atom for `addError` and `addFatal` accepting an Error object? That would be a nice shorthand for displaying errors, and I bet most people forget to include the stack, like here.

**Comment by florianb** [comment]
:rage: I guess i really have a problem with git-clients. This was my first (and last) PR with GitKraken.

File: spec/iss67-spec.js [hunk 2]
```diff
@@ -10,7 +10,9 @@ import {init as generateConfig} from '../commands/generate';
 
 describe('editorconfig', () => {
 	beforeEach(() => {
-		waitsForPromise(() => atom.packages.activatePackage('editorconfig'));
+		waitsForPromise(() => {
+			return atom.packages.activatePackage('editorconfig');
+		});
```
Comment (sindresorhus) [inline spec/iss67-spec.js]: This is exactly the same as the former, and the former is preferred.

File: commands/generate.js (hunk 1 above)
Comment (florianb) [inline commands/generate.js]: That's a good suggestion - i'll put it up.

File: spec/iss67-spec.js (hunk 2 above)
Comment (florianb) [inline spec/iss67-spec.js]: Really? i thought the return statement would be absolutely necessary. I'll rewind these kinds of lines.

**Comment by sindresorhus** [comment]
If you're on OS X, I would strongly recommend http://gitup.co ;)

File: spec/iss67-spec.js (hunk 2 above)
Comment (sindresorhus) [inline spec/iss67-spec.js]: Inline arrow functions implicitly returns. The previous and current has the exact same behavior.

Comment (florianb) [inline spec/iss67-spec.js]: Thanks.. sorry for acting that naive.

**Comment by florianb** [comment]
@sindresorhus - i rewrote the history (unfortunately gitup [which is really good -- 💯 ] set you as the author for https://github.com/sindresorhus/atom-editorconfig/pull/84/commits/93adadb3606c5a5ee965569859d64a729c2222b8)) and added the stacktrace as detail.

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "editorconfig",
-  "version": "1.3.0",
+  "version": "1.4.0",
```
Comment (sindresorhus) [inline package.json]: Don't bump the version in a PR. `apm` bumps it for me when doing a release ;)

**Comment by sindresorhus** [comment]
@florianb I merged and changed the author to you => https://github.com/sindresorhus/atom-editorconfig/commit/23b22995c7f933340a575e8ff32de186bd91992b

**Pull request closed without merging**